src/cluster.c

```diff
@@ -1471,12 +1471,20 @@ void askingCommand(client *c) {
  * In this mode replica will not redirect clients as long as clients access
  * with read-only commands to keys that are served by the replica's primary. */
 void readonlyCommand(client *c) {
+    if (server.cluster_enabled == 0 && !(c->capa & CLIENT_CAPA_REDIRECT)) {
```

Comment:
It feels weird to me to require sending the `CLIENT CAPA REDIRECT` first before readonly. I feel like clients should be able to send them in either order.

Overall i'm inclined to leave this as is.

Reply:
>It feels weird to me to require sending the CLIENT CAPA REDIRECT first before readonly. I feel like clients should be able to send them in either order.

It is not really about the order but more about compatibility, IMO. Without `CLIENT CAPA REDIRECT`, we fall back to the original behavior on standalone. 

 > This is also a breaking change from 8.0, which I don't want to do.
 
If we classify this as a bug, we should back port the fix to 8.0 and then the behavior would be the same across 7.2, 8.0, and future versions.